universal_transfer_operator/src/universal_transfer_operator/data_providers/filesystem/local.py

```diff
@@ -63,12 +65,14 @@ def check_if_exists(self) -> bool:
         """Return true if the dataset exists"""
         return exists(self.dataset.path)
 
-    def write_using_smart_open(self, source_ref: FileStream):
+    def write_using_smart_open(self, source_ref: FileStream | pd.DataFrame) -> str:
```

Comment:
```suggestion
    def write_using_stream(self, source_ref: FileStream | pd.DataFrame) -> str:
```

Reply:
@tatiana  I like the existing name in comparison because of the following reasons:
1. We don't always use a stream to write in some cases we also use dataframe. 
2. Also, we have different ways to write like write_using_hooks() and write_using_smart_open(), and these methods are called in general write() method.
3. I think it's clear from the typing of params what is used as source for the write operator. 

WDYT?